hivemq-edge/src/frontend/src/extensions/datahub/designer/schema/SchemaNode.utils.ts

```diff
@@ -18,6 +19,25 @@ import { PolicyCheckErrors } from '@datahub/designer/validation.errors.ts'
 import { enumFromStringValue } from '@/utils/types.utils.ts'
 import { CANVAS_POSITION } from '@datahub/designer/checks.utils.ts'
 
+export const getSchemaFamilies = (items: Schema[]) => {
+  return items.reduce<Record<string, ResourceFamily>>((acc, schema) => {
+    if (acc[schema.id]) {
+      if (schema.version) acc[schema.id].versions.push(schema.version)
+    } else {
+      let description: string | undefined
+      try {
+        const schemaDefinition = JSON.parse(atob(schema.schemaDefinition))
+        description = schemaDefinition.description
+      } catch (e) {
+        /* empty */
```

Comment:
question: is empty correct here?

smells a bit like missing error handling

Reply:
I'm not dealing with the error deliberately. I'm only interested in extracting the description from the content. If it fails, it's undefined. The error handling is done at creation